tests/register/register.go

```diff
@@ -31,6 +38,68 @@ func init() {
 	Tests = make(map[TestType][]types.Test)
 }
 
+var count = 0
+
 func Register(tType TestType, t types.Test) {
 	Tests[tType] = append(Tests[tType], t)
+	// count += howmany(t.Config)
+	count++
+	fmt.Printf("count now at: %d\n", count)
+}
+
+// Finds version, creates Test structs for all compatible versions and registers them
+func RegisterTest(tType TestType, t types.Test) {
+	versions := [][]string{
+		{"0.0.0"},
+		{"1.0.0"},
+		{"2.0.0", "2.1.0", "2.2.0", "2.3.0-experimental"},
+	}
+
+	pattern := regexp.MustCompile("[0-9]{1}.[0-9]{1}.[0-9]{1}")
+	versionArr := pattern.FindAllStringSubmatch(t.Config, 1)
+	if len(versionArr) == 0 { // handles negative/general/invalid_version.go
+		Register(tType, t)
+		return
+	}
+	version := versionArr[0][0]
+
+	major_version, _ := strconv.Atoi(version[0:1])
```

Comment:
Rather than doing string conversions and handling the numbers like this why not use the semver library to parse the string. If you bump the vendor'd `go-semver` package you'd get the access to built-in equality & comparison functions (https://github.com/coreos/go-semver/blob/master/semver/semver.go#L154-L162).